### PR TITLE
Add a toggle to force 30FPS mode

### PR DESCRIPTION
**Pull request by zarroboogs**
Continuing from a conversation on Discord with jroweboy, this PR implements a temporary hack that forces games to run in 30FPS mode.

The hack can be turned on/off using a toggle under the Graphics settings (off by default).

This should allow testers to properly test some 30FPS games, such as PLGP/E or BotW (these have messed up audio/physics when running in 60FPS mode).

Once a proper solution to the framerate limiting issue is implemented, this can be reverted or closed.

This hack can also be expanded to allow for a custom FPS value if needed.

### CODE REVIEW

File: src/core/hle/service/nvflinger/nvflinger.cpp [hunk 1]
```diff
@@ -29,20 +30,24 @@ constexpr std::size_t SCREEN_REFRESH_RATE = 60;
 constexpr s64 frame_ticks = static_cast<s64>(Core::Timing::BASE_CLOCK_RATE / SCREEN_REFRESH_RATE);
 
 NVFlinger::NVFlinger(Core::Timing::CoreTiming& core_timing) : core_timing{core_timing} {
+    const auto ticks = Settings::values.force_30fps_mode
+                           ? static_cast<s64>(Core::Timing::BASE_CLOCK_RATE / 30)
```
Comment (bunnei) [inline src/core/hle/service/nvflinger/nvflinger.cpp]: nit, move this above with frame_ticks, and call it frame_ticks_30fps

File: src/core/hle/service/nvflinger/nvflinger.cpp [hunk 2]
```diff
@@ -29,20 +30,24 @@ constexpr std::size_t SCREEN_REFRESH_RATE = 60;
 constexpr s64 frame_ticks = static_cast<s64>(Core::Timing::BASE_CLOCK_RATE / SCREEN_REFRESH_RATE);
```
Comment (bunnei) [inline src/core/hle/service/nvflinger/nvflinger.cpp]: feel free to remove SCREEN_REFRESH_RATE and just inline the 60

**Comment by bunnei** [comment]
Removing the DO NOT MERGE, since this is mergable IMO

File: src/core/hle/service/nvflinger/nvflinger.cpp (hunk 1 above)
Comment (zarroboogs) [reply src/core/hle/service/nvflinger/nvflinger.cpp]: Done.

File: src/core/hle/service/nvflinger/nvflinger.cpp (hunk 2 above)
Comment (zarroboogs) [reply src/core/hle/service/nvflinger/nvflinger.cpp]: Done.

File: src/core/hle/service/nvflinger/nvflinger.cpp [hunk 3]
```diff
@@ -21,28 +21,31 @@
...
 
 namespace Service::NVFlinger {
 
-constexpr std::size_t SCREEN_REFRESH_RATE = 60;
-constexpr s64 frame_ticks = static_cast<s64>(Core::Timing::BASE_CLOCK_RATE / SCREEN_REFRESH_RATE);
+constexpr s64 frame_ticks = static_cast<s64>(Core::Timing::BASE_CLOCK_RATE / 60);
+constexpr s64 frame_ticks_30fps = static_cast<s64>(Core::Timing::BASE_CLOCK_RATE / 30);
 
 NVFlinger::NVFlinger(Core::Timing::CoreTiming& core_timing) : core_timing{core_timing} {
+    const auto ticks = Settings::values.force_30fps_mode ? frame_ticks_30fps : frame_ticks;
```
Comment (lioncash) [inline src/core/hle/service/nvflinger/nvflinger.cpp]: This should be moved closer to its usage (after the comment about scheduling the composition events.

Comment (zarroboogs) [reply src/core/hle/service/nvflinger/nvflinger.cpp]: Done. Due to the "hacky" nature of this PR I tried making as few changes as possible, so I just left everything where it was before. Maybe I shouldn't have done that... 😄

File: src/yuzu/configuration/configure_graphics.ui [hunk 4]
```diff
@@ -70,6 +70,13 @@
           </property>
          </widget>
         </item>
+        <item>
+         <widget class="QCheckBox" name="force_30fps_mode">
+          <property name="text">
+           <string>Force 30FPS mode</string>
```
Comment (lioncash) [inline src/yuzu/configuration/configure_graphics.ui]: One last minor nitpick that I missed from the initial review.

The 30 and FPS should probably have a space in between them, like `30 FPS`.

Comment (zarroboogs) [reply src/yuzu/configuration/configure_graphics.ui]: Done.

File: src/core/hle/service/nvflinger/nvflinger.cpp [hunk 5]
```diff
@@ -62,6 +65,7 @@ std::optional<u64> NVFlinger::OpenDisplay(std::string_view name) {
     const auto itr =
         std::find_if(displays.begin(), displays.end(),
                      [&](const VI::Display& display) { return display.GetName() == name; });
+
```
Comment (FearlessTobi) [inline src/core/hle/service/nvflinger/nvflinger.cpp]: ultra nit: why was this added?

File: src/yuzu/configuration/configure_graphics.ui [hunk 6]
```diff
@@ -70,6 +70,13 @@
           </property>
          </widget>
         </item>
+        <item>
+         <widget class="QCheckBox" name="force_30fps_mode">
+          <property name="text">
+           <string>Force 30 FPS mode</string>
```
Comment (slashiee) [inline src/yuzu/configuration/configure_graphics.ui]: Another nit: wouldn't it be better to just call it "Force 30 FPS"? AFAIK it's not really a mode...

**Comment by bunnei** [comment]
Looks fine to me - This could use some tweaks in the future, but is useful as is for now, and for the sake of minimizing longstanding canary-tagged PRs, I think we can merge this one.
